core/src/test/java/org/elasticsearch/index/search/MatchQueryIT.java

```diff
@@ -172,6 +172,7 @@ public void testPhrasePrefix() throws ExecutionException, InterruptedException {
         assertSearchHits(searchResponse, "1", "2", "3", "7", "8");
     }
 
```

Comment:
Regarding the failing tests, this is due to a change in Lucene 6.5.0. In 6.4 Queries with multi word synonyms create a GraphQuery with all possible paths. So if you have "foo, foo bar" and a query like "foo boo" it will produce two boolean queries packed in a GraphQuery:
`foo boo` OR `foo bar boo`
Since we have all paths we can apply the cutoff frequency to each but in 6.5 the query parser will now create a query:
`(+foo OR (+foo +bar)) AND boo`
So the multi synonyms is now treated as a single term. This is better in terms of performance but it breaks some tests because the logic has changed. We cannot apply cutoff frequency on multi terms synonyms anymore (they are treated as a single term). Same problem with `minimum_should_match`.
IMO we should just rewrite this test and remove the multi terms synonym.